### DISCUSSION TITLE
[Website] Release notes selection does not work correctly

### DISCUSSION THREAD

**Discussion by Marcono1234** [bcgit/bc-java#1667] [General]
First of all, your new redesigned website looks really great!

But it seems there are some issues with the Release Notes selection at https://www.bouncycastle.org/download/bouncy-castle-java/#release-notes:

1. When you search for a version, e.g. "1.76" it shows on the dropdown on the left "All releases" instead of the exact matching version number (and it also does not actually show all releases)
![Version dropdown screenshot](https://github.com/bcgit/bc-java/assets/11685886/0fc8668e-6679-401f-ad52-fa547a73dc72)
2. When you search for a version, for some versions it also shows unrelated other versions
E.g. for "1.77" it additionally shows the release notes for 1.78.1 (possibly because they mention "1.77"); maybe that is a bit irritating?
On the other hand I guess the current behavior might be useful to find something in the release notes description, which you cannot find otherwise since not all release notes are shown at once.
   - Maybe it would then be useful if the version dropdown on the left also had an input field (some UIs support this; or maybe some other solution) to make it easier to find a certain version, so you don't have to rely on the "Search" for that?
   - And maybe it would be clearer if the input hint for the Search field said something like "Search descriptions" or similar?
3. If the dropdown on the left shows "All releases" and you click on it to unselect it, the text changes to "Latest release", but the selection above still says "All releases"
![Version dropdown screenshot](https://github.com/bcgit/bc-java/assets/11685886/d32ec872-b2f0-4e77-83e2-37d4c7033e74)
4. The option for "All releases" is not available by default, so you cannot show all release notes at once.
Not sure if that is intentional or not; maybe showing all versions could cause performance issues.

----

The release notes for C# seem to be affected as well:
- https://www.bouncycastle.org/download/bouncy-castle-c/#release-notes
- https://www.bouncycastle.org/download/bouncy-castle-c-fips/#release-notes

**Comment by KarolinHem** [comment]
Great to hear that you like the redesigned website! 

And thanks for your feedback on the Release notes section! 

Yes, you are right, the search is done through the content of **all** release notes. Therefore, when you search for “1.77”, multiple release notes are shown. So, to find a specific release, it is better to select that in the “Releases” drop-down.

“All releases” might be slightly confusing, but marks that all release notes that contain the search term are shown as a result. 

We are aware the search and filtering are not perfect as-is. We will look into improving them and will keep you updated.
